x-pack/plugins/security_solution/public/notes/store/notes.slice.ts

```diff
@@ -129,21 +129,21 @@ export const fetchNotes = createAsyncThunk<
     sortField: string;
     sortOrder: string;
     filter: string;
-    userFilter: string;
+    createdByFilter: string;
     associatedFilter: AssociatedFilter;
     search: string;
   },
   {}
 >('notes/fetchNotes', async (args) => {
-  const { page, perPage, sortField, sortOrder, filter, userFilter, associatedFilter, search } =
+  const { page, perPage, sortField, sortOrder, filter, createdByFilter, associatedFilter, search } =
     args;
   const res = await fetchNotesApi({
```

Comment:
Sorry I could not see where are we handling the error returned from the server.

Reply:
what do you mean?

Reply:
haaaa I see. Indeed we're not handling this in the frontend at this time. I don't think it should be a big deal as we're basically pinging the same endpoint to retrieve the uid in the combobox that we use in the `getNotes` endpoint to filter notes by `createdBy` user.
The main reason I added the `throw new Error` on the endpoint was for people that would be using the api direclty (it's public). I didn't want the api to return all notes (that was the previous behavior) if a user isn't found...

Reply:
also thinking about this a bit more, I don't think we're having an issue introduced by this PR. None of our api in [this file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/notes/api/api.ts) are handling errors. Yes I added a `throw new Error` in this PR, but errors could have happened before and we were not catching them.
I agree that we should handle all the errors more gracefully but I think this could be done as a separate PR, that would tackle all the note endpoints. It seems to me that we should do this as part of [note api changes](https://github.com/elastic/kibana/issues/194770) happening in `9.0`.

Thoughts?

Reply:
>I agree that we should handle all the errors more gracefully but I think this could be done as a separate PR, that would tackle all the note endpoints. It seems to me that we should do this as part of https://github.com/elastic/kibana/issues/194770 happening in 9.0.

I agree that it is not in the scope of this PR but i think we do handle many errors at many places by showing the error toast. **Or it may be possible that Kibana's `search` API handles that error**.  Can you have a task to look how the system reacts when there is an error?

Thanks for explanation.

Reply:
yup, I will definitely take a look!